File: src/main/java/com/learnershi/rclasssocket/repository/ClassRoomUserRepository.java
package com.learnershi.rclasssocket.repository;

import com.learnershi.rclasssocket.entity.User;

import java.util.Collection;

public interface ClassRoomUserRepository {

    void addUser(String classRoomId, User user);

    User removeUser(String classRoomId, String userSeq);

    Collection<User> findByClassRoomId(String classRoomId);

    User findUserByClassRoomIdAndUserSeq(String classRoomId, String userSeq);
}

File: src/main/java/com/learnershi/rclasssocket/repository/ClassRoomUserRepository.kt
package com.learnershi.rclasssocket.repository

import com.learnershi.rclasssocket.entity.User

interface ClassRoomUserRepository {
    fun addUser(classRoomId: String, user: User)
    fun removeUser(classRoomId: String, userSeq: String): User?
    fun findByClassRoomId(classRoomId: String): Collection<User?>
    fun findUserByClassRoomIdAndUserSeq(classRoomId: String, userSeq: String): User?
}